platformio.ini

```diff
@@ -742,6 +742,8 @@ src_filter    = ${common.default_src_filter} +<src/HAL/STM32F1>
 lib_ignore    = SPI
 lib_deps      = ${common.lib_deps}
   SoftwareSerialM
+# Reduce space usage by using a more recent version of arduinoststm32-maple than the one published on platformio
+platform_packages = framework-arduinoststm32-maple@https://github.com/mathiasvr/Arduino_STM32/archive/2020.11.29-dev.zip
```

Comment:
need to be replaced by "@~3.10000" to avoid dependencies to "unsafe" github repos

Reply:
Yes I mentioned this in my earlier comment

Reply:
but.. doesnt build fine here...
.platformio/packages/framework-arduinoststm32-maple/STM32F1/cores/maple/wirish_types.h:65:9: error: unknown type name 'bool'
 typedef bool boolean;

someone removed the -std=c++ by default

`arm-none-eabi-gcc -o .pio/build/alfawise_U30/libe8f/U8glib-HAL/clib/u8g_com_api.c.o -c -Os -mcpu=cortex-m3 -mthumb -fsigned-char -fno-move-loop-invariants -fno-strict-aliasing --specs=nano.specs --specs=nosys.specs -MMD -MP -fmax-errors=5 -g -fmerge-all-constants -Wno-attributes -Wno-unused-function -Wno-unused-const-variable -Wno-builtin-macro-redefined -mcpu=cortex-m3 -mthumb -ffunction-sections -fdata-sections -nostdlib -march=armv7-m --param max-inline-insns-single=500 -Wall -Os -DPLATFORMIO=50003 -DSTM32F103xE -DSTM32F1 -DTARGET_STM32F1 -D__MARLIN_FIRMWARE__ -DHAVE_SW_SERIAL -DMCU_STM32F103VE -DSTM32F1xx -DDEBUG_LEVEL=1 -D__FILE__=__func__ -DU30 -DTS_V12 -DARDUINO=10808 -DARDUINO_ARCH_STM32F1 -DARDUINO_GENERIC_STM32F103V -DMCU_STM32F103VE -D__STM32F1__ -DBOARD_generic_stm32f103v -DF_CPU=72000000L -DVECT_TAB_ADDR=0x8000000 -DSERIAL_USB -DGENERIC_BOOTLOADER -USERIAL_USB `

Reply:
nvm, i was on last tagged version branch... seems ok on current bugfix

Reply:
I'm actually also getting this error when using `@~3.10000` even on bugfix.

Reply:
I found that when I switch versions I have to perform a PIO Clean, or it has weird errors involving includes. It seems to work after that.

@tpruvot I probably removed some compiler flags because some overlapped between different sources. Some come from platformio.ini, some from a python script, some from board files, etc.

Reply:
I am trying to make each environment have as few unique details as possible, so that an environment is mostly an `extends` statement and a `board`.

Reply:
I still can't make @~3.10000 work, but I have pushed it to see if it's just on my end.

Reply:
I think I fixed this by adding `-DARDUINO_ARCH_STM32`, not sure why this is needed.

Reply:
Try removing that flag and doing another build, just in case the only effect was to force a recompile of the platforms and libraries.

Reply:
Several files depends on this flag being set, it was defined by the previous version of `framework-arduinoststm32-maple` for all stm32f1 envs.